common/utils/check_no_of_nodes.yml

```diff
@@ -0,0 +1,31 @@
+---
+- block:
+
+   - name: Waiting for application pod reschedule (evict)
+     wait_for:
+       timeout: 50
+
+   - name: Checking the status of chaos injected node
+     shell: kubectl get nodes --no-headers | grep -vi 'master'
+     args:
+       executable: /bin/bash
+     register: node_names
+     until: "'{{ app_node.stdout }}' not in node_names.stdout"
+     retries: 40
+     delay: 10
+
+  when: injected_node_chaos == "yes"
+
+
+- block:
+
+   - name: Checking the no.of nodes
+     shell: kubectl get nodes --no-headers | grep -vi 'master' | grep -i 'Ready'
+     args:
+       executable: /bin/bash
+     register: node_count
+     until: node_count.stdout_lines|length == pre_chaos_node_count.stdout_lines|length
+     retries: 10
+     delay: 60
+
+  when: injected_node_chaos == "no"
```

Comment:
Looks like a case of test business logic flowing into a util ? For ex: util to check (Ready) node count is great. But `waits`, actions for `pre` & `post` chaos may still be in test. However, this is just an opinion.

Reply:
Modified the condition to check (Ready) node count.